### ISSUE TITLE
Cant build my Android project

### ISSUE THREAD

**Issue by BigPun86** [idehub/react-native-google-analytics-bridge#50]
THere was a similar issue which is closed so i just opened a new one (#31). I tried to force using play-services.analytics like u mentioned but didn´t help....

I have several modules which depends/uses play-service. I solved my issues before using this module like u just mentioned by using 8.+ but since i included this module i get this error and the build takes for ever:

````

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:dexDebug'.
> com.android.ide.common.process.ProcessException: org.gradle.process.internal.ExecException: Process 'command '/Library/Java/JavaVirtualMachines/jdk1.8.0_77.jdk/Contents/Home/bin/java'' finished with non-zero exit value 2

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 6 mins 57.146 secs
````

When adding force 8.3.0 i get following error:

````

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:packageAllDebugClassesForMultiDex'.
> java.util.zip.ZipException: duplicate entry: com/google/android/gms/internal/zzka.class

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.

BUILD FAILED

Total time: 21.022 secs
````

````

dependencies {
    compile("com.google.android.gms:play-services-analytics:8.3.0"){
        force = true
    }
    compile fileTree(dir: "libs", include: ["*.jar"])
    compile "com.android.support:appcompat-v7:23.0.1"
    compile "com.facebook.react:react-native:+"  // From node_modules
    compile project(':react-native-i18n')
    compile project(':RNMaterialKit')
    compile project(':react-native-facebook-login')
    compile project(":react-native-google-signin")
    compile project(':RNDeviceInfo')
    compile project(':RNAdMob')
    compile project(':react-native-contacts')
    compile project(':react-native-maps')
    compile project(':react-native-billing')
    compile project(':react-native-fileupload')
    compile project(':react-native-sqlite-storage')
    compile project(':react-native-image-picker')
    compile project(':RNMail')
    compile project(':react-native-share')
    compile project(':react-native-push-notification')
    compile 'com.google.android.gms:play-services-gcm:8.3.0' // <- Add this line
    compile project(':RNGcmAndroid')                         // <- Add this line
    compile project(':react-native-system-notification')     // <- Add this line
    compile project(':react-native-dialogs')
    compile project(':react-native-fabric')
    compile project(':RNLocation')
    compile project(':react-native-google-analytics-bridge')

    compile('com.crashlytics.sdk.android:crashlytics:2.5.5@aar') {
        transitive = true;
    }

    apply plugin: 'com.google.gms.google-services' // <--- this should be the last line

}

````

**Comment by ElenaDiaz** [comment]
Same problem here. 
The only difference is that I am trying to integrate FCM instead of GCM. It seems like google-services:9.0.0 is needed.
But when I force 
`    compile("com.google.android.gms:play-services-analytics:9.0.0"){
        force = true
    }`

I get:

![image](https://cloud.githubusercontent.com/assets/12825893/17027983/f247de02-4f5d-11e6-8717-61b8b46a52a2.png)

**Comment by ElenaDiaz** [comment]
@BigPun86 
Do you have in your android/build.gradle 

`dependencies {
        classpath 'com.google.gms:google-services:3.0.0'
}`
?

**Comment by cbrevik** [comment]
This module depends on the 8.x.x version of the play-services library. If you are using another which depends on 9.x.x then I am not surprised it is crashing.

I have written somewhat about this in the [README](https://github.com/idehub/react-native-google-analytics-bridge#android-build-problem). As I say there, the "force compile" part might not be a solution if several modules (more than two) are depending on conflicting versions. Or if they are depending on different major versions.

Which other modules are you using which depends on 9.x.x?

**Comment by BigPun86** [comment]
@ElenaDiaz I am using 2.1.0 and i have set everywhere i use play-services etc. to 8.+ which worked fine before including this module

**Comment by BigPun86** [comment]
@cbrevik i am not using any module which uses 9.x.x so far i can see, but i will check this immediately :P

**Comment by BigPun86** [comment]
Adding this to my app/build.gradle solved it

````
  dexOptions {
        incremental true
        javaMaxHeapSize "2048M"
    }
````

**Comment by ElenaDiaz** [comment]
@cbrevik thanks for your reply.
I am trying to integrate FCM instead of GCM, because we have FCM already working with IOS and it is ok, but in Android I find dependencies incompatibility.

To integrate FCM, I have to include 
```dependencies { classpath 'com.google.gms:google-services:3.0.0' } ```

And it requires google-services 9.0.0.

I confirm you, forcing play-services-analytics to version 9.0.0, the app crashes.

**Comment by ivpusic** [comment]
same problem here. I am using https://github.com/devfd/react-native-google-signin which requires 9.0.0+

**Comment by ElenaDiaz** [comment]
After change some things, I am getting now the following error:

``` [$ProjectDir]/node_modules/react-native-fcm/android/src/main/java/com/evollu/react/fcm/FIRMessagingModule.java:159: error: method does not override or implement a method from a supertype
    @Override
    ^
 ```
@cbrevik Maybe there is any mistake in the doc about override this method?

Because you difference between RN0.28+ and RN<=27, but both options are identical.

![image](https://cloud.githubusercontent.com/assets/12825893/17053889/1b12853c-4ffc-11e6-924d-d8820dd42eea.png)

Btw, I am using RN 0.28.0

**Comment by cbrevik** [comment]
Well, this is going to be a headache for people. I will probably have to update the dependency to 9.x.x (after reviewing exactly what the breaking changes are).

But for you all depending on different libraries for play-services; if only one of them (like this one at the moment) depends on 8.x.x, this change is going to break the build.

This will probably solve it for the rest of you, but @BigPun86 will not be able to update since he is using modules which requires play-services 8.x.x (unless the other modules are updated to 9.x.x as well).

**Comment by ivpusic** [comment]
packages should support current not lold versions anyway...if there is a package supporting some old version, it is good sign that it is time for update

**Comment by ElenaDiaz** [comment]
Thank you @cbrevik .

**Comment by KGALLET** [comment]
got the same problem, i think this is because `compile project(":react-native-google-signin")` using play-services:9+ because before using this module, didn't have the problem

**Comment by cbrevik** [comment]
Alright guys, the Android dependency has been updated to 9.+.

I've released this as a major version, currently at `3.0.0`

**Comment by Kowaio** [comment]
Hi guys,

Just a up on this bug because I'm struggling to build for android. Here's the error I got:
```
* What went wrong:
Execution failed for task ':app:transformResourcesWithMergeJavaResForDebug'.
> com.android.build.api.transform.TransformException: com.android.builder.packaging.DuplicateFileException: Duplicate files copied in APK META-INF/maven/com.squareup.okhttp3/okhttp/pom.properties
  	File1: /Users/KowaioMBP/.gradle/caches/modules-2/files-2.1/com.squareup.okhttp3/okhttp/3.4.1/c7c4f9e35c2fd5900da24f9872e3971801f08ce0/okhttp-3.4.1.jar
  	File2: /Users/KowaioMBP/Desktop/Betacie/betaseries-native/android/app/build/intermediates/exploded-aar/com.google.android.gms/play-services-basement/10.2.6/jars/classes.jar
```

I added some packaging options in the gradle file like this:
```
    packagingOptions {
      exclude 'META-INF/maven/com.squareup.okhttp/okhttp/pom.xml'
      exclude 'META-INF/maven/com.squareup.okhttp/okhttp/pom.properties'
      exclude 'META-INF/maven/com.squareup.okio/okio/pom.xml'
      exclude 'META-INF/maven/com.squareup.okio/okio/pom.properties'
    }
```

And please find the dependencies below:
```
dependencies {
    compile project(':react-native-google-analytics-bridge')
    compile project(':lottie-react-native')
    compile project(':react-native-image-picker')
    compile project(':react-native-fbsdk')
    compile project(':react-native-restart')
    compile project(':react-native-vector-icons')
    compile project(':react-native-linear-gradient')
    compile project(':react-native-fs')
    compile project(':react-native-device-info')
    compile project(':react-native-onesignal')
    compile fileTree(dir: "libs", include: ["*.jar"])
    compile "com.android.support:appcompat-v7:23.0.1"
    compile "com.facebook.react:react-native:+"  // From node_modules
    compile 'com.batch.android:batch-sdk:1.5+'
    compile 'com.google.android.gms:play-services-gcm:9.0.2'
    compile('com.crashlytics.sdk.android:crashlytics:2.6.3@aar') {
      transitive = true;
    }
}
```

If anybody got a clue it'd be great!

Thanks for your help!

Best,
